File: Project1/2217.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;
int main(void)
{
    int n;
    cin >> n;
    vector<int> ropes(n);
    for (int i=0; i<n; ++i)
    {
        cin >> ropes[i];
    }
    sort(ropes.begin(), ropes.end());

    int result = 0;
    for (int i=n; i>=0; --i)
    {
        result = max(result, (n - i) * ropes[i]);
    }
    cout << result;
}

File: Project1/2217.py
import sys
input = sys.stdin.readline
n = int(input().strip())
ropes = []
for i in range(n):
    ropes.append(int(input().strip()))
ropes.sort()
result = 0
sum_ropes = 0
for i in range(n-1, -1, -1):
    sum_ropes += ropes[i]
    result = max(result, ropes[i] * (n-1-i+1))
print(result)
